socketio_middleware/src/rooms.rs

```diff
@@ -33,35 +35,114 @@ impl ChannelPair {
 }
 
 pub fn join_channel_to_room(room_id: &str, channel_pair: ChannelPair) {
+    //TODO: why use remove, it is low performance.
```

Comment:
You're right, this should probably be a `get_mut` (see [here](https://docs.rs/chashmap/2.2.2/chashmap/struct.CHashMap.html#method.get_mut).) Be warned though, that this will create a write lock so this method _must_ be super fast.

Reply:
Yes, I tested for using get_mut, it will occur PING/PONG socket close.  Let it go.